pkg/query-service/app/traces/v3/query_builder.go

```diff
@@ -85,18 +85,17 @@ func getClickhouseTracesColumnDataTypeAndType(key v3.AttributeKey) (v3.Attribute
 }
 
 func enrichKeyWithMetadata(key v3.AttributeKey, keys map[string]v3.AttributeKey) v3.AttributeKey {
-	if key.Type == "" || key.DataType == "" {
```

Comment:
I guess what breaks with removing this is when there is an attribute different type or datatype, irrespective of what user explicitly selects their preferred type and datatype, we override? I am not sure how much of a real problem this is. I have come across any customer who had the same attribute name with a different type or data type.

Reply:
That might be an issue but only when user is using custom attribute key with same name as otel key. I haven't encountered any users with the same attribute name and different types. 
In logs too we are also overriding them.

Reply:
> That might be an issue but only when user is using custom attribute key with same name as otel key. I haven't encountered any users with the same attribute name and different types. In logs too we are also overriding them.

There are a few users with the same key but different type/data type and they are not necessarily because of the custom attribute collision with otel key. See https://github.com/SigNoz/signoz/issues/5150 & https://github.com/SigNoz/signoz/issues/5359; @nityanandagohain created the issue after a bug request from a customer and then oss user also added that they are impacted by it.

Reply:
> > That might be an issue but only when user is using custom attribute key with same name as otel key. I haven't encountered any users with the same attribute name and different types. In logs too we are also overriding them.
> 
> There are a few users with the same key but different type/data type and they are not necessarily because of the custom attribute collision with otel key. See #5150 & #5359; @nityanandagohain created the issue after a bug request from a customer and then oss user also added that they are impacted by it.

Yeah, I saw community user faced the issue.

Let's block this until we find a fix

Reply:
They are independent problems, aren't they? Could we modify this pull request to still honor the user's selection (assuming the request payload is accurate), but for the attributes we know have fixed columns, use those instead?